core/unit_test/TestViewIsAssignable.hpp

```diff
@@ -0,0 +1,115 @@
+#include <Kokkos_Core.hpp>
+
+namespace Test {
+namespace Impl {
+template <class ViewTypeDst, class ViewTypeSrc>
+struct TestAssignability {
+  using mapping_type =
+      Kokkos::Impl::ViewMapping<typename ViewTypeDst::traits,
+                                typename ViewTypeSrc::traits,
+                                typename ViewTypeDst::specialize>;
+
+  template <class MappingType>
+  static void try_assign(
+      ViewTypeDst& dst, ViewTypeSrc& src,
+      typename std::enable_if<MappingType::is_assignable>::type* = nullptr) {
+    dst = src;
+  }
+
+  template <class MappingType>
+  static void try_assign(
+      ViewTypeDst&, ViewTypeSrc&,
+      typename std::enable_if<!MappingType::is_assignable>::type* = nullptr) {}
+
+  template <class... Dimensions>
+  static void test(bool always, bool sometimes, Dimensions... dims) {
+    ViewTypeDst dst;
+    ViewTypeSrc src("SRC", dims...);
+
+    // bool is_always_assignable =
+    // Kokkos::is_always_assignable<ViewTypeDst,ViewTypeSrc>::value;
+    bool is_always_assignable = Kokkos::is_always_assignable(dst, src);
+    bool is_assignable        = Kokkos::is_assignable(dst, src);
+
+    // Print out if there is an error with typeid so you can just filter the
+    // output with c++filt -t to see which assignment causes the error.
+    if (is_always_assignable != always || is_assignable != sometimes)
+      printf(
+          "is_always_assignable: %i (%i), is_assignable: %i (%i) [ %s ] to [ "
+          "%s ]\n",
+          is_always_assignable ? 1 : 0, always ? 1 : 0, is_assignable ? 1 : 0,
```

Comment:
Why do you use a ternary?  Aren't boolean promoted to `int` by default?

Reply:
I don't thinks so for printf in all implementations.